src/Resources/js/views/slotIndex.ts

```diff
@@ -0,0 +1,68 @@
+import Switchery from "switchery.js";
+
+export default class SlotIndexView {
+    constructor() {
+        this.init();
+    }
+
+    private init() {
+        this.initSlotStateToggler();
+        $('#slot-table .table-data.slot-actions a.delete').on('click', this.handleDeleteSlot.bind(this));
+    }
+
+    private initSlotStateToggler() {
+        const self = this;
+        $('#slot-table .table-data.slot-actions input[type="checkbox"]').each(function (idx, checkbox) {
+            new Switchery(checkbox, {
+                size: 'small',
+                onChange: function (state: boolean) {
+                    const id = $(checkbox).data('id');
+                    self.handleToggleSlotState(id, state);
+                }
+            });
+        });
+    };
+
+    private handleToggleSlotState(id: string, state: boolean) {
+        const url = '/meal/slot/' + id + '/update-state';
+        $.post(url, {'disabled': (false === state ? '1' : '0')})
+            .fail(function () {
+                $('.alert').show();
+            });
+    };
+
+    private handleDeleteSlot(event: JQuery.TriggeredEvent) {
+        let self = this; // SlotIndexView
```

Comment:
It's obvious that it is SlotIndexView, isn't it?

Reply:
No,  the context (this) has been overridden in line 10.